src/app/shared/components/course-form/course-form.component.ts

```diff
@@ -7,111 +7,167 @@ import {
   ValidationErrors,
   ValidatorFn,
   Validators,
-} from "@angular/forms";
-import { Author } from "@app/shared/types/author.model";
-import { ButtonTypes } from "@app/shared/types/button.type";
-import { IconNames } from "@app/shared/types/icons.model";
-import { v4 as uuidv4 } from "uuid";
+} from '@angular/forms';
+import { Router } from '@angular/router';
+import { CoursesStoreService } from '@app/services/courses-store.service';
+import { Author } from '@app/shared/types/author.model';
+import { ButtonTypes } from '@app/shared/types/button.type';
+import { Course, CourseInfo } from '@app/shared/types/course.model';
+import { IconNames } from '@app/shared/types/icons.model';
+import { Subscription } from 'rxjs';
 
 @Component({
-  selector: "app-course-form",
-  templateUrl: "./course-form.component.html",
-  styleUrls: ["./course-form.component.scss"],
+  selector: 'app-course-form',
+  templateUrl: './course-form.component.html',
+  styleUrls: ['./course-form.component.scss'],
 })
-export class CourseFormComponent {
+export class CourseFormComponent implements OnInit, OnDestroy {
   ButtonTypes = ButtonTypes;
   IconNames = IconNames;
 
+  @Input() isUpdate = false;
+  @Input() courseInfo?: CourseInfo;
+
   courseForm!: FormGroup;
-  allAuthors: Author[] = [];
+  private subscription: Subscription | undefined;
+  private allAuthors: Author[] = [];
   nonCourseAuthors: Author[] = [];
   submitted = false;
   wrongCreation = false;
+  actionText = 'CREATE COURSE';
+
+  constructor(
+    public fb: FormBuilder,
+    private router: Router,
+    private courseStore: CoursesStoreService
+  ) {}
+
+  ngOnInit() {
+    if (this.isUpdate) this.updateInit();
+    else this.createInit();
+  }
 
-  constructor(public fb: FormBuilder) {
+  createInit() {
+    this.buildForm();
+    this.subscription = this.courseStore.authors$.subscribe(authors$ => {
```

Comment:
Notice that these lines are very similar to what is on lines from 62. Can you please reafctor?